search-app/src/cmr/search/services/parameters/converters/shapefile.clj

```diff
@@ -131,6 +178,20 @@
     (debug (format "CONDITIONS: %s" conditions))
     [conditions point-count]))
 
+(defn features->conditions
+  "Converts a list of features into a vector of SpatialConditions"
+  [features mime-type]
+  (validate-features features)
+  (let [iterator (.iterator features)]
+    (loop [conditions []]
+      (if (.hasNext iterator)
+        (let [feature (.next iterator)
+              [feature-conditions _] (feature->conditions feature (winding-opts mime-type))]
+          (if (> (count feature-conditions) 0)
+            (recur (conj conditions (gc/or-conds feature-conditions)))
+            (recur conditions)))
+        conditions))))
```

Comment:
The nested `if`s and `let`s make me wonder if this can be broken up or at least more thoroughly commented to make it more readable

Reply:
Added comments